### ISSUE TITLE
CMS146v3 double counting patients

### ISSUE THREAD

**Issue by kghandi** [OSEHRA/popHealth#29]
I am running
popHealth commit: 90277b
last tag: v4.0.0
bundle 2.6.0

The patient data set that I loaded into pophealth for this measure only includes 2 patients. But the Measure is reporting: IPP=4, NUMER=2, DENOM=4, DENEX=2

**Comment by petercyli** [comment]
Can you tell me which measure is giving this incorrect result?

**Comment by kghandi** [comment]
CMS146v3 is the one I am having issue with. 

More info following additional debugging:
I have selected two measures. CMS146v3, and CMS154v3. The patients Margie Fox, and Dana Park are provided by cypress for both tests. I happen to export separate QRDA 1 test files for each measure. So there is a QRDA 1 file for each of:

* CMS146v3 Dana Park
* CMS146v3 Margie Fox
* CMS154v3 Dana Park
* CMS154v3 Margie Fox

All four files are loaded into pophealth. The calculation for CMS154 is only looking at the two 154 files and gives correct results. CMS146 seems to be evaluating all 4 files resulting in double counting of patients, even though two of the files are meant for a different measure.


I am now wondering if it is coincidence that none of the other 8 measures I have been using happen to produce duplicate counts.

Is the measure calculation expected to only load the qrda files which have the measure uuid? Or are all qrda files processed for every measure?

**Comment by kghandi** [comment]
After further testing, I still suspect that Measure CMS146v3 can double count patients. This seams to happen in a number of situations. 

One situation is the one described in the previous msg involving a 2nd QRDA file for the same patient but a different Measure.

But even after I removed the extra QRDA files, I could still get extra counts. This seems to happen when the QRDA file has multiple Encounter entries some of which over lap. That is I think multiple encounters are satisfying the IPP criteria.

I wish I could provide more info, but I don't know how to step thru the map_fn in a debugger.

**Comment by petercyli** [comment]
Can you send the qrda-1 files that creates double counts on CMS146v3?  I can look into the database to see if it is creating a new set of patients even though they are of same patient id as previous import.
Thanks.

**Comment by kghandi** [comment]
Two files are attached. Try these along with CMS146, and CMS154. 

If you only load one of the files each measure computes 1/1. If you load
both files, CMS154 still computes 1/1, but CMS146 computes 2/2.  

Somehow CMS154 recognizes that these are the same patient. But CMS146
does not. 

On 2015-11-26 10:46, petercyli wrote: 

> Can you send the qrda-1 files that creates double counts on CMS146v3? I can look into the database to see if it is creating a new set of patients even though they are of same patient id as previous import.
> Thanks. 
> 
> --
> Reply to this email directly or view it on GitHub [1].

  

Links:
------
[1] https://github.com/OSEHRA/popHealth/issues/29#issuecomment-159945027

**Comment by kghandi** [comment]
It seems attaching zip or xml is not supported. How shall I send files?

**Comment by petercyli** [comment]
Can you send it directly to my email address at: lip@osehra.org?
Thanks.

**Comment by petercyli** [comment]
Sorry to take this long to respond.  I've been tied up with working on a RFP response.

Regarding the issue with the double counts on CMS146v3, it was caused by having duplicates encounter entries created by the importing of duplicate records from the second qrda-1 import.  The logic that was used to eliminate the duplicate didn't work as expected.  We are in the process of improving the logic to remove the duplicate entry.  As to why the CMS146v3 created double counts, I didn't delve deeply into the measure calculation logic.  Moreover, the measure logic resides in the measure bundle and the QME, which are managed by the Cypress project.  Instead of trying to fix the measure logic, I decided the better approach is to make sure that the data is clean.
